Source/WebKit/GPUProcess/graphics/WebGPU/RemoteDevice.messages.in

```diff
@@ -47,7 +46,7 @@ messages -> RemoteDevice NotRefCounted Stream {
 #if PLATFORM(COCOA) && ENABLE(VIDEO)
     void SetSharedVideoFrameSemaphore(IPC::Semaphore semaphore) NotStreamEncodable
     void SetSharedVideoFrameMemory(WebKit::SharedMemory::Handle storageHandle) NotStreamEncodable
-    void ImportExternalTextureFromPixelBuffer(WebKit::WebGPU::ExternalTextureDescriptor descriptor, std::optional<WebKit::SharedVideoFrame::Buffer> sharedBuffer, WebKit::WebGPUIdentifier identifier) NotStreamEncodable
+    void ImportExternalTextureFromVideoFrame(WebKit::WebGPU::ExternalTextureDescriptor descriptor, std::optional<WebKit::SharedVideoFrame> sharedFrame, WebKit::WebGPUIdentifier identifier) NotStreamEncodable
```

Comment:
I think this is yet to be addressed, I will take another look now

Reply:
`sharedFrame` is not part of `ExternalTextureDescriptor` because its definition as an RValue in SharedVideoFrame.serialization.in appears to prevent it from being used in a struct with StreamConnectionEncoder.

But maybe I don't even need the descriptor and can just pass parameters. I will try that.

Reply:
Passing all the information in the descriptor via parameters is somewhat fragile since this descriptor derives from a base type and any change there would require updated parameters.

Maybe @achristensen07 knows if we can use the auto-generated serialization for structs containing members of types like `SharedVideoFrame` which use the Rvalue specifier like so?
```
[RValue] struct WebKit::SharedVideoFrame
```

Reply:
I forgot to add the containing type to `types_that_must_be_moved`, with that change it seems to work correctly now